packages/wrangler/turbo.json

```diff
@@ -58,12 +61,97 @@
 			]
 		},
 		"test:ci": {
-			"inputs": ["!**/.wrangler/**", "**/__tests__/**"],
-			"dependsOn": ["build"]
+			"inputs": ["!*/**/.wrangler/**", "**/__tests__/**"],
+			"dependsOn": ["build"],
+			"env": [
+				"TMPDIR",
+				"TMP",
+				"TEMP",
+				"NODE_EXTRA_CA_CERTS",
+				"CLOUDFLARE_ACCOUNT_ID",
+				"SOURCEMAPS",
+				"NODE_ENV",
+				"SPARROW_SOURCE_KEY",
+				"ALGOLIA_APP_ID",
+				"ALGOLIA_PUBLIC_KEY",
+				"NO_SCRIPT_SIZE_WARNING",
+				"CLOUDFLARE_API_TOKEN",
+				"CLOUDFLARE_ACCOUNT_ID",
+				"WRANGLER_AUTH_DOMAIN",
+				"PATH",
+				"WRANGLER_LOG",
+				"EXPERIMENTAL_MIDDLEWARE",
+				"FORMAT_WRANGLER_ERRORS",
+				"CF_PAGES",
+				"CI",
+				"CF_PAGES_UPLOAD_JWT",
+				"NO_SCRIPT_SIZE_WARNING",
+				"EXPERIMENTAL_MIDDLEWARE",
+				"NO_D1_WARNING",
+				"NO_HYPERDRIVE_WARNING",
+				"WRANGLER",
+				"WRANGLER_IMPORT",
+				"CUSTOM_BUILD_VAR",
+				"PWD",
+				"LC_ALL",
+				"WRANGLER_SEND_METRICS",
+				"https_proxy",
+				"HTTPS_PROXY",
+				"http_proxy",
+				"HTTP_PROXY",
+				"CI_OS",
+				"SENTRY_DSN",
+				"SYSTEMROOT",
+				"WRANGLER_DISABLE_EXPERIMENTAL_WARNING",
+				"WRANGLER_DISABLE_REQUEST_BODY_DRAINING",
+				"WRANGLER_WORKER_REGISTRY_PORT",
+				"WRANGLER_API_ENVIRONMENT"
+			]
 		},
 		"test:e2e": {
 			"inputs": ["e2e/**"],
-			"dependsOn": ["build"]
+			"dependsOn": ["build"],
+			"env": [
+				"NODE_EXTRA_CA_CERTS",
+				"CLOUDFLARE_ACCOUNT_ID",
+				"SOURCEMAPS",
+				"NODE_ENV",
+				"SPARROW_SOURCE_KEY",
+				"ALGOLIA_APP_ID",
+				"ALGOLIA_PUBLIC_KEY",
+				"NO_SCRIPT_SIZE_WARNING",
+				"CLOUDFLARE_API_TOKEN",
+				"CLOUDFLARE_ACCOUNT_ID",
+				"WRANGLER_AUTH_DOMAIN",
+				"PATH",
+				"WRANGLER_LOG",
+				"EXPERIMENTAL_MIDDLEWARE",
+				"FORMAT_WRANGLER_ERRORS",
+				"CF_PAGES",
+				"CI",
+				"CF_PAGES_UPLOAD_JWT",
+				"NO_SCRIPT_SIZE_WARNING",
+				"EXPERIMENTAL_MIDDLEWARE",
+				"NO_D1_WARNING",
+				"NO_HYPERDRIVE_WARNING",
+				"WRANGLER",
+				"WRANGLER_IMPORT",
+				"CUSTOM_BUILD_VAR",
+				"PWD",
+				"LC_ALL",
+				"WRANGLER_SEND_METRICS",
+				"https_proxy",
+				"HTTPS_PROXY",
+				"http_proxy",
+				"HTTP_PROXY",
+				"CI_OS",
+				"SENTRY_DSN",
+				"SYSTEMROOT",
```

Comment:
All these feel unnecessary. If they are only in the source files then they are already covered by the `build` task. Here we should only be declaring the env vars that are referenced in the input files (i.e. `"e2e/**"`).

Reply:
If they're read dynamically, then they have to be whitelisted. I considered being more precise, but that would make this more confusing for the future. Instead, I duplicate them across tasks because it's easier to make changes in the future too (where the instruction will be "hey just add it to both"